src/Database/Esqueleto/Legacy.hs

```diff
@@ -130,7 +131,7 @@ module Database.Esqueleto.Legacy
   ) where
 
 import Database.Esqueleto.Internal.Internal
-import Database.Esqueleto.Internal.PersistentImport
+import Database.Esqueleto.Internal.PersistentImport hiding (selectFirst)
```

Comment:
Same issue here (and elsewhere) - let's keep this an open import and hide the export at the source.
```suggestion
import Database.Esqueleto.Internal.PersistentImport
```